### PR TITLE
feat(metrics): emit a login_complete glean event on login completion

### PR DESCRIPTION
**Pull request by chenba**

### CODE REVIEW

File: packages/fxa-auth-server/lib/metrics/events.js [hunk 1]
```diff
@@ -140,6 +140,10 @@ module.exports = (log, config) => {
         });
         await amplitude('flow.complete', request, data, metricsContext);
 
+        if (metricsContext.flowType === 'login') {
+          glean.login.complete(request, { uid: data?.uid ?? '' });
+        }
```
Comment (chenba) [inline packages/fxa-auth-server/lib/metrics/events.js:145]: The signal's registered at https://github.com/mozilla/fxa/blob/1386b2de515ddc0a078083ff1d7e3ebde2f05484/packages/fxa-auth-server/lib/routes/utils/signin.js#L228-L241 (which include some relevant comments).

**Review by vpomerleau — APPROVED** [review]
LGTM. Tested locally with regular login and sync and I see the events coming through as expected. Just one question - what is the difference with `login_success`? I see both pings going through at the same time. Is there a case where we would see one but not the other?

**Comment by chenba** [comment]
> Just one question - what is the difference with `login_success`? 

I think `login_success` is when you entered the correct password during login, but login_complete is at teh end of the flow (after 2fa, session confirmation via code, or whatever the extra steps might be).

**Comment by vpomerleau** [comment]
> > Just one question - what is the difference with `login_success`?
> 
> I think `login_success` is when you entered the correct password during login, but login_complete is at teh end of the flow (after 2fa, session confirmation via code, or whatever the extra steps might be).

Gotcha, just looks simultaneous then? I was a bit puzzled looking at the logs because `login_complete` is sometimes logged before `login_success` on regular logins (not sync, not must verify) 🤔

**Comment by chenba** [comment]
> > > Just one question - what is the difference with `login_success`?
> > 
> > 
> > I think `login_success` is when you entered the correct password during login, but login_complete is at teh end of the flow (after 2fa, session confirmation via code, or whatever the extra steps might be).
> 
> Gotcha, just looks simultaneous then? I was a bit puzzled looking at the logs because `login_complete` is sometimes logged before `login_success` on regular logins (not sync, not must verify) 🤔

It might the async nature of the execution.  I don't think we await those calls in some kind of order?

**Pull request merged**